### PR TITLE
docs: executor crate

### PR DESCRIPTION
**Pull request by 0xKitsune**
I am opening up this draft PR to track progress on the docs for the executor crate.

### CODE REVIEW

File: crates/executor/src/executor.rs [hunk 1]
```diff
@@ -149,7 +149,7 @@ pub fn commit_changes<DB: StateProvider>(
             db_account.account_state = AccountState::NotExisting;
             db_account.info = AccountInfo::default();
 
-            continue
+            continue;
```
Comment (gakonst) [inline crates/executor/src/executor.rs:152]: pls remove code chagnes

Comment (0xKitsune) [reply crates/executor/src/executor.rs]: Thanks for pointing this out! It looks like my "format on save" was enabled. I made sure to turn it off so this doesn't happen in future PRs. Sorry for the mixup! Everything should be fixed now and the only changes should be within `docs/repo/crates/executor.md`.

**Comment by jinsankim** [comment]
https://github.com/paradigmxyz/reth/pull/1096 

As a newbie, I'm working on `executor` even though it's just a draft. I'm afraid that my works impacts on this work because it contains too detail flow.

**Comment by gakonst** [comment]
closing as this is unfortunately outdated per the last refactor. @0xKitsune I think any future docs let's try to keep them on the structs vs in separate markdown files

**Comment by 0xKitsune** [comment]
> I think any future docs let's try to keep them on the structs vs in separate markdown files.

Sounds good I will keep this in mind for the next section of docs that I work on.

**Comment by gakonst** [comment]
Thanks boss and appreciate the understanding

**Pull request closed without merging**